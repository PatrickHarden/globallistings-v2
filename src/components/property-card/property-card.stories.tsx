import React from 'react';
import { storiesOf } from '@storybook/react';
import { PropertyCard } from './property-card';

storiesOf('Property Card', module)
  .add('Primary style', () => (
    <PropertyCard
      address={'88 Londond Wall, EC2R 5BT'}
      imgUrl={'https://images.unsplash.com/photo-1485181044355-5b94cc167214'}
      price={'£42.50/sqft/pa'}
      size={'3,102sqft - 8,102sqft'}
      id={'1'}
      location={{ lat: 1.123, lon: 1.123 }}
      selected={false}
    />
  ))
  .add('With click event', () => (
    <PropertyCard
      address={'88 Londond Wall, EC2R 5BT'}
      imgUrl={'https://images.unsplash.com/photo-1485181044355-5b94cc167214'}
      price={'£42.50/sqft/pa'}
      size={'3,102sqft - 8,102sqft'}
      id={'1'}
      location={{ lat: 1.123, lon: 1.123 }}
      handleClick={(id: string) => console.log(`property ${id} clicked`)}
      selected={false}
    />
  ));
