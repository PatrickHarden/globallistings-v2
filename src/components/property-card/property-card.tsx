import React, { CSSProperties } from 'react';
import styled from 'styled-components';
import { PropertyOverview } from '../../redux/selectors/property-search-results';
import { ResponsiveImage } from '../responsive-image/responsive-image';

interface CardContainerProps {
  readonly clickable: boolean;
  readonly selected: boolean;
}

const CardContainer = styled.div<CardContainerProps>`
  background: ${({ theme }) => theme.colours.light};
  display: flex;
  border: ${({ theme, selected }) => (selected ? theme.borders.thick : theme.borders.fine)};
  box-shadow: ${({ theme }) => theme.shadow.light};
  height: 122px;
  cursor: ${({ clickable }) => (clickable ? 'pointer' : 'auto')};
`;

const ResponsiveImageStyled = styled(ResponsiveImage)`
  width: 140px;
  max-height: 100%;
  flex-shrink: 0;
`;

const CardInfo = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${({ theme }) => theme.gutter.s};
  justify-content: space-between;
`;

const AddressLine = styled.span`
  margin-bottom: ${({ theme }) => theme.gutter.s};
`;

const PriceAndSizeContainer = styled.div`
  display: flex;
  flex-direction: column;
  font-size: ${({ theme }) => theme.font.size.s};
`;

const Price = styled.span`
  color: ${({ theme }) => theme.colours.primary};
  margin-bottom: ${({ theme }) => theme.gutter.s};
`;

const Size = styled.span``;

export type Props = PropertyOverview & {
  className?: string;
  handleClick?: (id: string) => void;
  handleMouseEnter?: (id: string) => void;
  handleMouseLeave?: (id: string) => void;
  style?: CSSProperties;
};

export const PropertyCard: React.SFC<Props> = ({
  address,
  className,
  handleClick,
  handleMouseEnter,
  handleMouseLeave,
  id,
  imgUrl,
  price,
  selected,
  size,
  style,
}) => (
  <CardContainer
    className={className}
    clickable={!!handleClick}
    onClick={() => handleClick && handleClick(id)}
    onMouseEnter={() => handleMouseEnter && handleMouseEnter(id)}
    onMouseLeave={() => handleMouseLeave && handleMouseLeave(id)}
    selected={selected}
    style={style}
  >
    <ResponsiveImageStyled src={imgUrl} />
    <CardInfo>
      <AddressLine>{address}</AddressLine>

      <PriceAndSizeContainer>
        <Price>{price}</Price>
        <Size>{size}</Size>
      </PriceAndSizeContainer>
    </CardInfo>
  </CardContainer>
);
