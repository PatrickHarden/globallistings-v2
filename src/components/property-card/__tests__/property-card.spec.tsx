import React from 'react';
import { shallow } from 'enzyme';

import { PropertyCard } from '../property-card';

describe(PropertyCard, () => {
  const props = {
    address: '1 Any Road, Any WHere',
    id: 'abc123',
    imgUrl: '/img',
    location: { lat: 0, lon: 0 },
    price: 'Â£100',
    size: 'Massive',
    selected: false,
  };

  it('should render correctly', () => {
    expect(shallow(<PropertyCard {...props} />)).toMatchSnapshot();
  });

  it('should render correctly with a click event', () => {
    expect(shallow(<PropertyCard {...props} handleClick={id => id} />)).toMatchSnapshot();
  });
});
