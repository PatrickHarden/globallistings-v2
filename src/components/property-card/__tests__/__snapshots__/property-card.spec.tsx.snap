// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard should render correctly 1`] = `
<styled.div
  clickable={false}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  selected={false}
>
  <Styled(ResponsiveImage)
    src="/img"
  />
  <styled.div>
    <styled.span>
      1 Any Road, Any WHere
    </styled.span>
    <styled.div>
      <styled.span>
        £100
      </styled.span>
      <styled.span>
        Massive
      </styled.span>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`PropertyCard should render correctly with a click event 1`] = `
<styled.div
  clickable={true}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  selected={false}
>
  <Styled(ResponsiveImage)
    src="/img"
  />
  <styled.div>
    <styled.span>
      1 Any Road, Any WHere
    </styled.span>
    <styled.div>
      <styled.span>
        £100
      </styled.span>
      <styled.span>
        Massive
      </styled.span>
    </styled.div>
  </styled.div>
</styled.div>
`;
